rfd/0216-tbot-resiliency.md

```diff
@@ -0,0 +1,214 @@
+---
+authors: Dan Upton (daniel.upton@goteleport.com)
+state: draft
+---
+
+# RFD 216 - Improving `tbot`'s Resiliency for Production Workloads
+
+## What
+
+Making `tbot` more resilient to auth server availability issues, with graceful
+degradation.
+
+## Why
+
+As customers move from using Machine and Workload Identity in CI/CD pipelines to
+putting us in the critical path of their "online" systems (e.g. by using SPIFFE to
+secure service-to-service communication), the risk and blast radius of a Teleport
+outage increases dramatically.
+
+Wherever it is practical and safe to do so, `tbot` should gracefully handle
+transient issues, and fall back to using stale or cached data rather than
+failing completely - which could impact the availability of customer applications.
```

Comment:
I think it's worth listing the services we expect to benefit from this and the limitation of those benefits:
- Application Tunnel:
  - Potentially limited benefit when auth server is offline at tbot startup.
  - Would allow existing app tunnel connections to survive the auth server going offline for extended periods.
- Database Tunnel:
  - Potentially limited benefit when auth server is offline at tbot startup.
  - Would allow existing db tunnel connections to survive the auth server going offline for extended periods.
- AWS Roles Anywhere:
  - Allows the service to run if credentials are cached and the auth server is offline at tbot startup.
  - Allows the service to continue to run using the cached credentials if the auth server goes offline whilst tbot is running.